### ISSUE TITLE
False positive "Unresolved reference": multiple choices.

### ISSUE THREAD

**Issue by alexander-irbis** [intellij-rust/intellij-rust#3083]
## Environment

* **Intellij-Rust plugin version:** v0.2.0.2111-183
* **Rust toolchain version:** stable-x86_64-apple-darwin (default)
    rustc 1.30.1 (1433507eb 2018-11-07)
* **IDE name and version:** IntelliJ IDEA 2018.3 (Ultimate Edition)
    Build #IU-183.4284.148, built on November 21, 2018
    JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
* **Operating system:** macOS 10.14.1

## Problem description

![default](https://user-images.githubusercontent.com/1448323/49077410-e09e4700-f243-11e8-8e00-5340dd7ff98f.png)
![default](https://user-images.githubusercontent.com/1448323/49077486-0deaf500-f244-11e8-9efb-a118035659bb.png)

## Steps to reproduce
Correct runnable example:
```rust
use std::fmt;

struct X {
    x: String,
}

impl fmt::Display for X {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        self.x.fmt(f)
    }
}

impl fmt::Debug for X {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        self.x.fmt(f)
    }
}

fn main() {
    let x = X { x: "123".to_string() };
    println!("Display: {}", x);
    println!("Debug: {:?}", x);
}
```
Output:
```
Display: 123
Debug: "123"
```

**Comment by alexander-irbis** [comment]
Such a code, on the contrary, is not detected as erroneous, but Rust rejects it:
![default](https://user-images.githubusercontent.com/1448323/49078473-c87bf700-f246-11e8-8473-c6dd58a16d07.png)
```rust
use std::fmt;
use std::fmt::Display;
use std::fmt::Debug;

struct X {
    x: String,
}

impl Display for X {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        self.x.fmt(f)
    }
}

impl Debug for X {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        self.x.fmt(f)
    }
}

fn main() {
    let x = X { x: "123".to_string() };
    println!("Display: {}", x);
    println!("Debug: {:?}", x);
}
```

```
error[E0034]: multiple applicable items in scope                                                                                                                                                     
  --> .../examples/display.rs:11:16                                                                                                                                                                 
   |                                                                                                                                                                                                 
11 |         self.x.fmt(f)                                                                                                                                                                           
   |                ^^^ multiple `fmt` found                                                                                                                                                         
   |                                                                                                                                                                                                 
   = note: candidate #1 is defined in an impl of the trait `std::fmt::Display` for the type `std::string::String`                                                                                    
   = note: candidate #2 is defined in an impl of the trait `std::fmt::Debug` for the type `std::string::String`                                                                                      
                                                                                                                                                                                                     
error[E0034]: multiple applicable items in scope                                                                                                                                                     
  --> .../examples/display.rs:17:16                                                                                                                                                                 
   |                                                                                                                                                                                                 
17 |         self.x.fmt(f)                                                                                                                                                                           
   |                ^^^ multiple `fmt` found                                                                                                                                                         
   |                                                                                                                                                                                                 
   = note: candidate #1 is defined in an impl of the trait `std::fmt::Display` for the type `std::string::String`                                                                                    
   = note: candidate #2 is defined in an impl of the trait `std::fmt::Debug` for the type `std::string::String`                                                                                      
                                                                                                                                                                                                     
error: aborting due to 2 previous errors                                                                                                                                                             
```

**Comment by vlad20012** [comment]
Fixes by #3698
